statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+
+        # The total number of clusters over all stratum
+        self.nclust = np.sum(self.ncs)
+
+    def __str__(self):
+        """
+        The __str__ method for our data
+        """
+        summary_list = ["Number of observations: ", str(len(self.strat)),
+                        "Sum of weights: ", str(self.weights.sum()),
+                        "Number of strata: ", str(self.nstrat),
+                        "Number of clusters per stratum:", str(self.ncs)]
+
+        return "\n".join(summary_list)
+        # print("Number of observations: ", len(self.strat))
+        # print("Sum of weight: ", self.weights.sum())
+        # print("Number of strata: ", self.nstrat)
+        # print("The number of clusters per stratum: ", self.ncs)
+
+    def _check_args(self, strata, cluster, weights):
+        """
+        Minor error checking to make sure user supplied any of
+        strata, cluster, or weights. For unspecified subgroup labels
+        an array of ones is created
+
+        Parameters
+        ----------
+        strata : array-like or None
+            Strata for each observation. If none, an array
+            of ones is constructed
+        cluster : array-like or None
+            Cluster for each observation. If none, an array
+            of ones is constructed
+        weights : array-like or None
+            The weight for each observation. If none, an array
+            of ones is constructed
+
+        Returns
+        -------
+        vals[0] : ndarray
+            array of the strata labels
+        vals[1] : ndarray
+            array of the cluster labels
+        vals[2] : ndarray
+            array of the observation weights
+        """
+        if all([x is None for x in (strata, cluster, weights)]):
+            raise ValueError("""At least one of strata, cluster, and weights
+                             musts not be None""")
+        v = [len(x) for x in (strata, cluster, weights) if x is not None]
+        if len(set(v)) != 1:
+            raise ValueError("""lengths of strata, cluster, and weights
+                             are not compatible""")
+        n = v[0]
+        vals = []
+        for x in (strata, cluster, weights):
+            if x is None:
+                vals.append(np.ones(n))
+            else:
+                vals.append(np.asarray(x))
+
+        return vals[0], vals[1], vals[2]
+
+
+class SurveyStat(object):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, mse):
+        self.design = design
+        self.mse = mse
+
+    def bootstrap(self, replicates, index=None):
```

Comment:
When we first discussed the quantile, we decided that it should have the column index as a parameter. Thus, when I get it's se via bootstrap or jackknife, I call it in a for loop for each column index. This is so that jdata isnt 3d (num clusters x num percentiles x num columns)